### PR TITLE
KAZOO-3904: As a webhooks user, I would like a webhook that notifies me any time a doc changes

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

**Comment by fenollp** [comment]
@macpie 
```
warn_matching                 webhooks_util.erl:539: The pattern {'error', _} can never match the type boolean()
warn_matching                 webhooks_util.erl:540: The pattern {'ok', Doc} can never match the type boolean()
warn_not_called               webhooks_util.erl:541: Function will never be called
warn_not_called               webhooks_util.erl:551: Function load_metadata/2 will never be called
warn_not_called               webhooks_util.erl:555: Function will never be called
warn_not_called               webhooks_util.erl:557: Function will never be called
warn_not_called               webhooks_util.erl:559: Function will never be called
warn_not_called               webhooks_util.erl:566: Function update_metadata/2 will never be called
```

**Comment by k-anderson** [comment]
@macpie I actually think I liked objects better :(  Lets discuss this with the team.

**Comment by macpie** [comment]
I think objects does not make sense as it is a doc update your are subscribing for

File: applications/webhooks/src/webhooks_maintenance.erl [hunk 1]
```diff
@@ -110,3 +111,20 @@ enable_account_hooks(AccountId) ->
 
 enable_descendant_hooks(AccountId) ->
     webhooks_util:reenable(AccountId, <<"descendants">>).
+
+-spec clean_webhooks_list() -> 'ok'.
+clean_webhooks_list() ->
```
Comment (jamesaimonetti) [inline applications/webhooks/src/webhooks_maintenance.erl]: Seems like this is resetting the listing, not cleaning

Comment (macpie) [inline applications/webhooks/src/webhooks_maintenance.erl]: Indeed renaming function

**Comment by jamesaimonetti** [comment]
@macpie `doc` is a couchdb concept that we don't really extend into Kazoo. We name things `JObj` instead of `JDoc`, we are renaming `kazoo_documents` to `kazoo_objects`, etc.

**Comment by macpie** [comment]
OK I get that but you are still  subscribing for couch db doc update

**Comment by jamesaimonetti** [comment]
@macpie There are other "objects" that aren't tied to documents in couch. Plus, changing the module name makes it hard to see what you've actually changed! So, since there doesn't seem to be a compelling reason to rename object to doc, let's revert it back so we can better look over the PR and get it merged.

**Comment by macpie** [comment]
ready

**Comment by jamesaimonetti** [comment]
@k-anderson HALP? What happened here? I rebased origin/master and...it took a shit all over the PR.

**Comment by jamesaimonetti** [comment]
@k-anderson I think I figured it out. Nothing to see here

**Pull request merged**